### PR TITLE
docs: update rebasing documentation

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

This is in essence a copy/paste of the good stuff from PR #8029.

- Reduce capitalization
- Better explain that Renovate "rebase" is not Git rebase
- Fix white space in ordered list
- Let Markdown renderer worry about ordering the list
- Rewrite manual rebasing section

## Context:

Let's try this again in a docs only format... 😄 

Helps with #7626.

The Renovate wizards will need to check if the descriptions are still accurate to the current Renovate back-end behavior, as that was recently changed.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
🧙🧙 😅

File: docs/usage/updating-rebasing.md [hunk 1]
```diff
@@ -1,38 +1,46 @@
...
 
-## Rebasing Out-of-date Branches
+## Rebasing out-of-date branches
 
 There are two cases where Renovate will rebase its branches off the base branch every time they are out of date:
 
-1.  If you configure `"rebaseWhen": "behind-base-branch"`
-2.  If you use the default configuration `"rebaseWhen": "auto"` and the repository has a requirement that branches must be up-to-date before merging (e.g. "Require branches to be up to date before merging" on GitHub, or FF-only settings on Bitbucket Server or GitLab)
+1. If you configure `"rebaseWhen": "behind-base-branch"`
+1. If you use the default configuration `"rebaseWhen": "auto"` and the repository has a requirement that branches must be up-to-date before merging (e.g. "Require branches to be up to date before merging" on GitHub, or FF-only settings on Bitbucket Server or GitLab)
```
Comment (HonkingGoose) [inline docs/usage/updating-rebasing.md:39]:
```suggestion
1. If you use the default configuration `"rebaseWhen": "auto"` and the repository has a requirement that branches must be up-to-date before merging (e.g. "Require branches to be up to date before merging" on GitHub, or fast-forward-only settings on Bitbucket Server or GitLab)
```

How about this? I think `fast-forward-only` is clearer than a somewhat cryptic `FF-only`.

**Comment by HonkingGoose** [comment]
> 🧙🧙 😅

@viceice Just a bit of off-topic fun, see if you can figure this one out... 😉 

🔥 🔥 🔥 👹 🔥 🔥 🔥
🌉
🧙 

🧙 ✨ ✨ ✨ 

🌉 💥 

👹 ⬇️ 

👹 ⚡ 🧙 

🧙 ⬇️

**Comment by HonkingGoose** [comment]
How do we want to deal with this in the docs?
Copy/pasted from https://github.com/renovatebot/renovate/discussions/8982#discussioncomment-439745

> 
> 
> Hey guys,
> 
> good I found this. If this is new behaviour the docs on this are a little misleading I think:
> 
> https://docs.renovatebot.com/updating-rebasing/#rebasing-out-of-date-branches
> 
> I would think that renovate will not by default rebase every merge request, when something else is committed to the master branch. In my case I definitely have to disable this setting and use the `"rebaseWhen": "conflicted"`, because I have multiple CI Pipelines with e2e tests end everything, which run up to 5 times before they get finally merged, because they play ping pong with other merge requests.
> 
> But now I know about the new default I'm happy :)
> 
> Thanks a lot for renovate guys, saves me hours every week!

**Comment by rarkins** [comment]
> I would think that renovate will not by default rebase every merge request, when something else is committed to the master branch.

This is either not correct or a misunderstanding. And now it's clearly documented when it *will* happen (which is not every time, by default).

I don't see any further action required.

**Comment by HonkingGoose** [comment]
@rarkins Okay, just wanted to check as we're editing the file now anyways. 😄

File: docs/usage/updating-rebasing.md [hunk 2]
```diff
@@ -1,38 +1,46 @@
...
 
-Note: Renovate doesn't technically do "rebasing" in the Git sense.
-Instead, it reapplies all updates into a new commit based off of the head of the base branch.
+Here is a list of the most common cases where Renovate must update/rebase the branch:
+
+- When a pull request has conflicts due to changes on the base branch
+- When you have enabled "Require branches to be up to date before merging" on GitHub
+- When you have manually told Renovate to rebase when behind the base branch with `"rebaseWhen": "behind-base-branch"`
+- When a newer version of the dependency is released
+- When you request a manual rebase from the Renovate bot
```
Comment (viceice) [inline docs/usage/updating-rebasing.md:16]:
```suggestion
- When you use `automerge` on a branch / pr
```

File: docs/usage/updating-rebasing.md [hunk 3]
```diff
@@ -1,38 +1,46 @@
...
+You can disable this functionality by configuring `"rebaseWhen": "never"` (not recommended).
 
-## Rebasing Out-of-date Branches
+## Rebasing out-of-date branches
 
 There are two cases where Renovate will rebase its branches off the base branch every time they are out of date:
 
 1. If you configure `"rebaseWhen": "behind-base-branch"`
-2. If you use the default configuration `"rebaseWhen": "auto"` and the repository has a requirement that branches must be up-to-date before merging (e.g. "Require branches to be up to date before merging" on GitHub, or FF-only settings on Bitbucket Server or GitLab)
+1. If you use the default configuration `"rebaseWhen": "auto"` and the repository has a requirement that branches must be up-to-date before merging (e.g. "Require branches to be up to date before merging" on GitHub, or fast-forward-only settings on Bitbucket Server or GitLab)
```
Comment (viceice) [inline docs/usage/updating-rebasing.md:39]:
```suggestion
1. If you use the default configuration `"rebaseWhen": "auto"` and the repository has a requirement that branches must be up-to-date before merging (e.g. "Require branches to be up to date before merging" on GitHub, or fast-forward-only settings on Bitbucket Server or GitLab)
1. If you configure `automerge=true` ...
```

🙃

File: docs/usage/updating-rebasing.md [hunk 4]
```diff
@@ -1,38 +1,48 @@
...
+1. If you configure `automerge=true` ...
```
Comment (HonkingGoose) [inline docs/usage/updating-rebasing.md:41]: L41 is still a bit vague right now:

```
1. If you configure `automerge=true` ...
```

Do we want to explain more here, or do we just want to that using `automerge=true` is one of the cases where Renovate will rebase its branches off the base branch.

File: docs/usage/updating-rebasing.md [hunk 5]
```diff
@@ -1,38 +1,48 @@
...
 If new commits to the base branch - such as merging another Renovate PR - result in an open Renovate PR having merge conflicts, then Renovate will recreate ("rebase") any conflicted PRs.
 This applies both to commits to dependency files such as `package.json` as well as lock files such as `yarn.lock`.
 You should not ever need to resolve such conflicts manually.
-You can disable this functionality by configuring `"rebaseWhen": "never"` (not recommended);
+You can disable this functionality by configuring `"rebaseWhen": "never"` (not recommended).
 
-## Rebasing Out-of-date Branches
+## Rebasing out-of-date branches
 
 There are two cases where Renovate will rebase its branches off the base branch every time they are out of date:
```
Comment (HonkingGoose) [inline docs/usage/updating-rebasing.md:37]:
```suggestion
There are multiple cases where Renovate will rebase its branches off the base branch every time they are out of date:
```

File: docs/usage/updating-rebasing.md (hunk 4 above)
Comment (viceice) [reply docs/usage/updating-rebasing.md:41]: yes, if branch is out of date and rebaseWhen is auto -> renovate will rebase. was just a starter to complete by you 😉

File: docs/usage/updating-rebasing.md [hunk 6]
```diff
@@ -1,38 +1,48 @@
...
-Note: Renovate doesn't technically do "rebasing" in the Git sense.
-Instead, it reapplies all updates into a new commit based off of the head of the base branch.
+Here is a list of the most common cases where Renovate must update/rebase the branch:
+
+- When a pull request has conflicts due to changes on the base branch
+- When you have enabled "Require branches to be up to date before merging" on GitHub
+- When you have manually told Renovate to rebase when behind the base branch with `"rebaseWhen": "behind-base-branch"`
+- When a newer version of the dependency is released
+- When you request a manual rebase from the Renovate bot
+- When you use `automerge` on a branch / pr
```
Comment (viceice) [inline docs/usage/updating-rebasing.md:17]:
```suggestion
- When you use `"automerge": true` and `"rebaseWhen": "auto"` on a branch / pr
```

Is this clear enough ?

File: docs/usage/updating-rebasing.md (hunk 4 above)
Comment (HonkingGoose) [reply docs/usage/updating-rebasing.md]: Can you take a look at the new L41 and tell me what you think?

**Review by viceice — APPROVED** [review]
LGTM

**Comment by HonkingGoose** [comment]
@rarkins What do you think of this?

**Pull request merged**